<?xml version="1.0"?>
<launch>
   <arg name="video" default="0" />
   <arg name="video_width" default="640" />
   <arg name="video_height" default="480" />
   <arg name="video_visualize" default="false" />
   <arg name="orbslam_setting_path" default="$(find orb_slam2)/conf/fpvcam.yaml" />

   <include file="$(find video_stream_opencv)/launch/camera.launch" required="true">
        <!-- node name and ros graph name -->
        <arg name="camera_name" value="videofile" />
        <!-- means video device 0, /dev/video0 -->
        <arg name="video_stream_provider" value="$(arg video)" />
        <!-- throttling the querying of frames to -->
        <arg name="fps" value="30" />
        <!-- setting frame_id -->
        <arg name="frame_id" value="videofile_frame" />
        <!-- camera info loading, take care as it needs the "file:///" at the start , e.g.:
        "file:///$(find your_camera_package)/config/your_camera.yaml" -->
        <arg name="camera_info_url" value="" />
        <!-- flip the image horizontally (mirror it) -->
        <arg name="flip_horizontal" value="false" />
        <!-- flip the image vertically -->
        <arg name="flip_vertical" value="false" />
        <!-- force a width and height, 0 means no forcing -->
        <arg name="width" value="$(arg video_width)"/>
        <arg name="height" value="$(arg video_height)"/>
        <!-- visualize on an image_view window the stream generated -->
        <arg name="visualize" value="$(arg video_visualize)" />
    </include>

    <node name="orb_slam2" pkg="orb_slam2" type="orb_slam2_mono" output="screen" required="true">
        <param name="camera" value="/videofile/image_raw" />
        <param name="path_vocabulary" value="$(env ORBSLAM2_HOME)/Vocabulary/ORBvoc.txt" />
        <param name="path_settings" value="$(arg orbslam_setting_path)" />
        <param name="use_viewer" type="bool" value="true" />
    </node>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen" required="true">
        <remap from="image" to="/videofile/image_raw"/>
        <param name="autosize" value="true" />
    </node>


    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg cf_frame) /crazyflie/base_link 100" />
    <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos.rviz" required="true" />

</launch>
